src/app/features/matching-entities/matching-entities/matching-entities.component.ts

```diff
@@ -186,4 +209,43 @@ export class MatchingEntitiesComponent
   applySelectedFilters(side: MatchingSide, filter: DataFilter<Entity>) {
     side.filterObj = Object.assign({}, filter, side.prefilter ?? {});
   }
+
+  entityInMapClicked(entity: Entity) {
+    const side = this.sideDetails.find(
+      (s) => s.entityType === entity.getConstructor()
+    );
+    if (side) {
+      side.selectMatch(entity);
+    }
+  }
+
+  private initDistanceColumn() {
+    this.columns?.forEach((side) => {
+      if (side.includes(this.distanceColumn.id)) {
+        const index = side.indexOf(this.distanceColumn.id);
+        side[index] = this.distanceColumn;
+      }
+    });
+  }
+
+  private calculateDistanceTo(e: Entity) {
+    // TODO does not update with selection
+    if (this.leftSide.selected && this.leftSide.selected !== e) {
+      return this.getDistanceString(this.leftSide.selected, e, 0);
+    } else if (this.rightSide.selected && this.rightSide.selected !== e) {
+      return this.getDistanceString(this.rightSide.selected, e, 1);
+    } else {
+      return "-";
+    }
+  }
+
+  private getDistanceString(from: Entity, to: Entity, index: number): string {
+    const a = from[this.showMap[index]] as Coordinates;
+    const b = to[this.showMap[(index + 1) % 2]] as Coordinates;
+    if (!a || !b) {
+      return "-";
+    }
+    const res = getKmDistance(a, b).toFixed(2);
+    return $localize`:distance with unit|e.g. 5 km:${res} km`;
+  }
```

Comment:
Is there any feasible way to move the whole distance column logic to its own component? Maybe as a special ViewComponent? This would help keep the MatchingComponent more manageable, as it would be only about the immediate matching logic.

Reply:
I don't know if it was worth. The `MatchingComponent` only got a few lines shorter but at least the distance is always correctly up-to-date.